presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -263,45 +261,29 @@ protected RoleGrant computeNext()
         });
     }
 
-    public static boolean isRoleApplicable(SemiTransactionalHiveMetastore metastore, HivePrincipal principal, String role)
+    public static boolean isRoleApplicable(HivePrincipal principal, String role, Function<HivePrincipal, Set<RoleGrant>> listRoleGrants)
```

Comment:
Maybe we pass `SqlStandardAccessControlMetadataMetastore` here, instead of a function?

although, i see the value of this too

Reply:
Yeah - it would work. Using `Function` here was kinda an intermediate step. I squash this with `Interface out SqlStandardAccessControlMetadataMetastore`

Reply:
Naa - I take that back. Let's keep the function. Going with `SqlStandardAccessControMetastore` implies that I need to also pass `ConnectoSecurityContext` in `isRoleApplicable` signature. It does not make it substantially simpler